test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -1946,30 +1947,43 @@ def mkdtemp(self, hostname=None, suffix='', prefix='PtlPbs', dir=None,
                      directory
         :param level: logging level, defaults to INFOCLI2
         """
+        # create a temp dir as current user
+        tmpdir = tempfile.mkdtemp(suffix, prefix, dir)
```

Comment:
you might not have permission to create dir under user requested dir. I suggest you create dir under tmp and then move it to requested dir as given user.